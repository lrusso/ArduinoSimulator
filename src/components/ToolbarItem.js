import React from "react"

const ToolbarItem = ({ onClick, children }) => {
  return (
    <div style={styles.item}>
      <div
        style={styles.iconWrapper}
        onClick={onClick}
        className="arduinosimulator_menu_item"
      >
        {children}
      </div>
    </div>
  )
}

const styles = {
  item: {
    float: "left",
    display: "flex",
    margin: 0,
    height: "40px",
    alignItems: "center",
  },
  iconWrapper: {
    display: "block",
    fontFamily: "Arial",
    fontSize: "15px",
    lineHeight: "32px",
    height: "28px",
    width: "28px",
    backgroundColor: "#F2F2F2",
    border: "thin solid #F2F2F2",
    marginLeft: "3px",
    cursor: "default",
    WebkitUserSelect: "none",
    MozUserSelect: "none",
    userSelect: "none",
    paddingTop: "3px",
    paddingBottom: "1px",
  },
}

export default ToolbarItem
