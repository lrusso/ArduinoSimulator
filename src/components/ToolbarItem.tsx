import React from "react"

interface ToolbarItemProps {
  onClick: () => void
  disabled?: boolean
  children: React.ReactNode
}

const ToolbarItem = ({ onClick, disabled, children }: ToolbarItemProps) => {
  return (
    <div style={styles.item}>
      <div
        style={styles.iconWrapper}
        onClick={onClick}
        className={disabled ? undefined : "arduinosimulator_menu_item"}
      >
        {children}
      </div>
    </div>
  )
}

const styles: { [key: string]: React.CSSProperties } = {
  item: {
    float: "left",
    display: "flex",
    margin: 0,
    height: "40px",
    alignItems: "center",
  },
  iconWrapper: {
    display: "block",
    fontFamily: "Arial",
    fontSize: "15px",
    lineHeight: "32px",
    height: "28px",
    minWidth: "28px",
    backgroundColor: "#181818",   
    marginLeft: "3px",
    cursor: "default",
    WebkitUserSelect: "none",
    MozUserSelect: "none",
    userSelect: "none",
    paddingTop: "3px",
    paddingBottom: "1px",
    fill: "#868686",
  },
}

export default ToolbarItem
