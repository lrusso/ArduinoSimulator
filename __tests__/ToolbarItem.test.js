import React from "react"
import renderer from "react-test-renderer"
import ToolbarItem from "../src/components/ToolbarItem"

describe("ToolbarItem", () => {
  test("Rendering 1", () => {
    const tree = renderer
      .create(
        <ToolbarItem>
          <></>
        </ToolbarItem>
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})
