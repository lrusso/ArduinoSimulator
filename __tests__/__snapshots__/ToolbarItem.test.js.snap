// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ToolbarItem Rendering 1 1`] = `
<div
  style={
    {
      "alignItems": "center",
      "display": "flex",
      "float": "left",
      "height": "40px",
      "margin": 0,
    }
  }
>
  <div
    className="arduinosimulator_menu_item"
    style={
      {
        "MozUserSelect": "none",
        "WebkitUserSelect": "none",
        "backgroundColor": "#F2F2F2",
        "border": "thin solid #F2F2F2",
        "cursor": "default",
        "display": "block",
        "fontFamily": "Arial",
        "fontSize": "15px",
        "height": "28px",
        "lineHeight": "32px",
        "marginLeft": "3px",
        "paddingBottom": "1px",
        "paddingTop": "3px",
        "userSelect": "none",
        "width": "28px",
      }
    }
  />
</div>
`;
